app/models/publishers/employer_enrollment_notification.rb

```diff
@@ -0,0 +1,95 @@
+module Publishers
+  class EmployerEnrollmentNotification
+
+    PublishError = Struct.new(:message, :headers)
+
+    attr_reader :employer
+
+    def initialize(employer)
+      @employer = employer
+    end
+
+    def employer_policies
+
+      policies = Policy.where( :employer_id => employer.id,
+                               :aasm_state.in => %w[submitted resubmitted effectuated],
+                               :enrollees =>
+                                   { "$elemMatch" => {"$or" => [
+                                       {"coverage_end" => nil},
+                                       {"coverage_end" => { "$gt" => Time.now }}]
+                                   }
+                               })
+      policies.select{|pol| (pol.is_active? || pol.future_active?) && pol.carrier.shop_profile.requires_employer_updates_on_enrollments }
```

Comment:
Your life will be easier if you put the carrier ID in the where by looking up carriers which support this functionality, and then searching the policy using a $where $in for the carrier ID.  That way you don't have to load the policy to exclude it.

You may also want to put some sort of lower bound on the start date of the policy, like a year or so.

Reply:
@TreyE updated policy query to look for carrier id and added boundary for coverage_start date